### PR TITLE
[stable/docker-registry] Add pod annotations and checksum for configmap to deployment

### PR DESCRIPTION
**Pull request by rendhalver**
The will allow us to annotate the pod as necessicary and will restart the Deployemnt on ConfigMap changes

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: stable/docker-registry/values.yaml [hunk 1]
```diff
@@ -9,6 +9,8 @@ updateStrategy:
   #   maxSurge: 1
   #   maxUnavailable: 0
 
+podAnnotations: {}
+
```
Comment (rjkernick) [inline stable/docker-registry/values.yaml:13]: Please document this in the readme

**Review by rjkernick — CHANGES_REQUESTED** [review]

File: stable/docker-registry/values.yaml (hunk 1 above)
Comment (rendhalver) [reply stable/docker-registry/values.yaml:13]: Good point.
I will add that.

Comment (rendhalver) [reply stable/docker-registry/values.yaml:13]: I have added this to the README.

**Comment by rendhalver** [comment]
@unguiculus There was a PR merged that updated the docker-registry so I had to rebase.
Can I get another review please?

**Pull request merged**